ElectronicObserverTypes/Extensions/FleetDataExtensions.cs

```diff
@@ -108,4 +109,105 @@ ShipTypes.Battlecruiser or
 
 		return heavyCruisers > 3;
 	}
+
+	/// <summary>
+	/// https://x.com/yukicacoon/status/1739480992090632669
+	/// </summary>
+	/// <param name="fleets"></param>
+	/// <returns></returns>
+	public static List<SmokeGeneratorTriggerRate> GetSmokeTriggerRates(this List<IFleetData> fleets)
+	{
+		if (fleets.Count is 0) return [];
+
+		IShipData? flagship = fleets[0].MembersWithoutEscaped?.First(s => s?.IsFlagship(fleets[0]) is true);
```

Comment:
```cs
fleets[0].MembersWithoutEscaped?.First()
```
That should be enough here. I actually have no idea why the IsFlagship function exists.